### PR TITLE
feat: flag "use to" when "used to" is intended

### PR DESCRIPTION
**Pull request by ghost**
## Summary

Adds a phrase set correction that flags "use to" and suggests "used to" when meaning accustomed to (after forms of `be` or `get`).

## Patterns covered

- Be verbs + "use to" (am/is/are/was/were/be use to)
- Be contractions + "use to" (I'm/we're/you're/they're/he's/she's/it's use to)
- Getting/get/got + "use to"

Pronoun patterns (I/we/they use to) were intentionally excluded to avoid false positives with legitimate verb usage like "the editor I use to write code".

## Test plan

- `cargo test -p harper-core --lib -- phrase_set_corrections::tests` - 3 true positives + 2 true negatives pass
- Correct "used to" is not flagged
- "use to" as a verb is not flagged

Partial fix for #2969 (covers be/get patterns; pronoun patterns need POS-aware approach)

### CODE REVIEW

**Comment by elijah-potter** [comment]
It looks like some snapshot tests failed. Don't worry — there is an easy fix.

Assuming you have Rust installed, just run `cargo test || cargo test` to regenerate the snapshot files, then commit the resulting changes.

Thanks for opening a PR!

File: harper-core/src/linting/phrase_set_corrections/mod.rs [hunk 1]
```diff
@@ -514,6 +514,29 @@ pub fn lint_group() -> LintGroup {
             "Ensures proper use of the subjunctive mood in counterfactual conditional statements starting with `if only` or `I wish`.",
             LintKind::Grammar
         ),
+        "UseToUsedTo" => (
+            &[
+                ("i use to", "i used to"),
+                ("we use to", "we used to"),
+                ("they use to", "they used to"),
+                ("you use to", "you used to"),
```
Comment (hippietrail) [inline harper-core/src/linting/phrase_set_corrections/mod.rs]: These context-free ones will match things like "This is the editor * use to write code". Maybe more often that matching the actual mistake.

File: harper-core/src/linting/phrase_set_corrections/mod.rs [hunk 2]
```diff
@@ -514,6 +514,29 @@ pub fn lint_group() -> LintGroup {
...
+                ("he use to", "he used to"),
+                ("she use to", "she used to"),
```
Comment (hippietrail) [inline harper-core/src/linting/phrase_set_corrections/mod.rs]: These will match things like "What editor does he/she use to write code?" Also likely to hit more false positives that actual mistakes.

File: harper-core/src/linting/phrase_set_corrections/mod.rs [hunk 3]
```diff
@@ -514,6 +514,29 @@ pub fn lint_group() -> LintGroup {
...
+                ("it use to", "it used to"),
```
Comment (hippietrail) [inline harper-core/src/linting/phrase_set_corrections/mod.rs]: "What API does this app use to get that effect?" etc.

File: harper-core/src/linting/phrase_set_corrections/mod.rs [hunk 4]
```diff
@@ -514,6 +514,29 @@ pub fn lint_group() -> LintGroup {
...
+                ("getting use to", "getting used to"),
+                ("get use to", "get used to"),
+                ("got use to", "got used to"),
+                ("am use to", "am used to"),
+                ("are use to", "are used to"),
+                ("is use to", "is used to"),
+                ("was use to", "was used to"),
+                ("were use to", "were used to"),
+                ("be use to", "be used to"),
```
Comment (hippietrail) [inline harper-core/src/linting/phrase_set_corrections/mod.rs:525]: For these ones false positives aren't springing to mind, but I always do tons of Google searches to check because language is full of surprises.

Contractions of be would also fit:
I'm use to, we're use to, you're use to, he's use to, she's use to, it's use to, they're use to.

File: harper-core/tests/text/linters/The Great Gatsby.snap.yml [hunk 5]
```diff
@@ -5363,6 +5363,16 @@ Suggest:
 
 
 
+Lint:    Grammar (31 priority)
+Message: |
+    4066 | “That was his cousin. I knew his whole family history before he left. He gave me
+    4067 | an aluminum putter that I use to-day.”
```
Comment (hippietrail) [inline harper-core/tests/text/linters/The Great Gatsby.snap.yml]: This is an edge case with a very old-fashioned spelling of today. It can be prevented by checking for following hyphen. I know two ways to do that in an `ExprLinter` but `phrase_set` can't do it and I'm not sure if Weir can do it but I'm assuming not.

File: harper-core/src/linting/phrase_set_corrections/mod.rs (hunk 1 above)
Comment (ghost) [reply harper-core/src/linting/phrase_set_corrections/mod.rs]: You're right, dropped all the pronoun patterns. They're way too likely to hit legitimate verb usage. Kept only the be-verb and get-verb patterns where it's unambiguously wrong, and added contractions too (I'm/we're/etc). Also added a false positive test for the verb case.

File: harper-core/src/linting/phrase_set_corrections/mod.rs (hunk 4 above)
Comment (ghost) [reply harper-core/src/linting/phrase_set_corrections/mod.rs:525]: Added contractions: I'm/we're/you're/they're/he's/she's/it's use to. These should all be safe since a contraction of be + use to is always wrong.

File: harper-core/tests/text/linters/The Great Gatsby.snap.yml (hunk 5 above)
Comment (ghost) [reply harper-core/tests/text/linters/The Great Gatsby.snap.yml]: Good catch on the old-fashioned spelling. Since phrase_set can't check for a following hyphen, this is a known limitation for now. If it becomes a problem in practice we could convert this specific pattern to an ExprLinter later.

**Comment by ghost** [comment]
Good idea, updated the PR description. Changed it to reference #2969 without closing it so the harder pronoun cases stay open for someone to tackle with a POS-aware approach.

File: harper-core/src/linting/phrase_set_corrections/mod.rs (hunk 1 above)
Comment (hippietrail) [reply harper-core/src/linting/phrase_set_corrections/mod.rs]: I also maintain a corpus of real-word coding-related text that I often run linters I'm working on over. It helps to discover edge cases where you never would've expected any false positives. Mine are from the trending page of GitHub that I use to download their READMEs. There's probably a ton of ways to do it.

Comment (ghost) [reply harper-core/src/linting/phrase_set_corrections/mod.rs]: That's a great idea, thanks for the tip. Running against real README text would definitely catch edge cases that unit tests miss.
